### PR TITLE
Add Heroku easy deploy and Social login optionals

### PR DESCRIPTION
**Pull request by nicolasleger**
To allow project to be easily reusable by others.

- ENV Settings list in `README.md`
- Social login as optional according to presence of ENV `FACEBOOK_CLIENT_ID` and/or `GOOGLE_CLIENT_ID` for each, test by toggling each one

- Heroku one-click free deploy with addons via `app.json` to be able to reuse the platform for own purpose or test
    Test : https://heroku.com/deploy?template=https://github.com/nicolasleger/covoiturage-libre/tree/add_heroku_easy_deploy
Live demo : https://covoiturage-libre.herokuapp.com

- Scalingo one-click deploy with addons via `scalingo.json` to be able to reuse the platform for own purpose or test
    Test : https://my.scalingo.com/deploy?source=https://github.com/nicolasleger/covoiturage-libre#add_heroku_easy_deploy

### CODE REVIEW

**Comment by Poussinou** [comment]
Ouuuah!
Hi @nicolasleger thanks for the PR! First of all, sorry for the time to respond, we are all volunteers in the team. :)

For the project, can you maybe developp a little bit more what your PR does? We will test it and then push this PR if all is OK on our test-website;)

Please @pimpin @didierclin @flexbox @Alex131089 can you test this PR on your own, and report any problem here before I push? If you want me to remove you from the list of persons I can ask to test the PRs, just ask;)

You can join the test group on facebook here : https://www.facebook.com/groups/covoiturage.libre.test/?fref=ts

Again, thanks for the PR, we want you to developp the website :1st_place_medal: 

Ask us if you have any question

@thb can you look a little deeper into this PR?

**Comment by nicolasleger** [comment]
I updated the PR description to explain changes and how to test it.

File: README.md [hunk 1]
```diff
@@ -75,6 +75,10 @@ Then you can run the migrations
 - `SERVER_NAME` : optional, for redirects only
 - `SLAASK_WIDGET_KEY` : optional, for visitors chat with you
 
+## Enable Postgis on PostgreSQL
```
Comment (pimpin) [inline README.md]: This duplicates postgis setup.
Either your note should be at l42, or you should move l42..49 here.
I'm not sure about any requiered order between involved steps, but I'd rather keep the existing order.

Comment (nicolasleger) [inline README.md]: You're right, I updated the PR, thanks!

**Comment by nicolasleger** [comment]
@Poussinou I added Scalingo easy deploy, you can test via https://my.scalingo.com/deploy?source=https://github.com/nicolasleger/covoiturage-libre#add_heroku_easy_deploy

**Comment by webplate** [comment]
This seems very powerful :) I'm not a cloud user but might become !
Congrats @nicolasleger !!
Please could you make a wiki page detailing the ENV variables ? This would be a nice reference to have for newcomers.
@Poussinou anyway I can't wait to see our beautiful PRs merged !!

**Comment by nicolasleger** [comment]
Thanks @webplate I already started to list `ENV` variables in the the ReadMe.
We should list them in one place only, so in ReadMe or in Wiki?

**Comment by Poussinou** [comment]
Next week, I think I will have the capacities and the time to merge and try all the PRs :)
@nicolasleger please open a new page in the wiki and list all the variables here

**Comment by webplate** [comment]
@nicolasleger I've tried your branch like this https://my.scalingo.com/deploy?source=https://github.com/nicolasleger/covoiturage-libre#add_heroku_easy_deploy
and got error 500 :(
log:
https://framabin.org/?173e31312aec07dd#mWvNEdrbYEaoPABwIo5dFifCB2IxdzDM3Icz1KEEdpo=

are the cms pages loaded correctly ?

**Comment by nicolasleger** [comment]
@Poussinou I created and copied ENV variables configuration to https://github.com/covoiturage-libre/covoiturage-libre/wiki/Configuration-ENV-variables redirecting ReadMe to it

**Comment by nicolasleger** [comment]
@webplate thanks for your feedback,
I didn't get the error via the deploy link, no action excepted I filled " localhost " to all ENV settings, demo at https://testccc.scalingo.io/ with log available at https://framabin.org/?d3c2e6f1ab44e82d#QWElg6MyhZcCfHxDV/hUy+ostkgdmz7d86WgeC1apiE=

Can you retry and submit the whole deploy and building log from Scalingo ? (seems the migration was not made but I can't figured it out why)

@Poussinou : We might remove optional setting keys from the Scalingo deploy since all are taken as required ?

**Comment by webplate** [comment]
Ok it actually works when submitting _localhost_ to all ENV variables :)
Maybe this should be specified somewhere ?

**Comment by nicolasleger** [comment]
Some are mandatory some not, but Scalingo takes all as required whereas Heroku, we should describe them in the new Wiki page.

File: app/views/layouts/application.html.haml [hunk 2]
```diff
@@ -21,8 +21,10 @@
     %meta{property: 'og:site_name',    content: 'Covoiturage-libre.fr'}
     %meta{property: 'og:image',        content: @meta[:image_url]}
     %meta{property: 'og:url',          content: @meta[:url]}
-    / facebook app id
-    %meta{property: 'fb:app_id',       content: ENV['FACEBOOK_CLIENT_ID']}
+
+    - if ENV['FACEBOOK_CLIENT_ID']
```
Comment (thb) [inline app/views/layouts/application.html.haml]: Actually this is not related to Facebook login system, but to the Facebook application needed for opengraph meta tags (necessary for social & SSO)

Comment (nicolasleger) [inline app/views/layouts/application.html.haml]: @thb sorry, i don't get the point, OpenGraph info are still above if you do not have a `FACEBOOK_CLIENT_ID`

**Pull request merged**